src/status_im/subs/chats.cljs

```diff
@@ -8,20 +8,11 @@
     [status-im.contexts.profile.utils :as profile.utils]
     [status-im.subs.contact.utils :as contact.utils]))
 
-(def memo-chats-stack-items (atom nil))
-
-(re-frame/reg-sub
- :chats/chats-stack-items
+(re-frame/reg-sub :chats/chats-stack-items
  :<- [:chats/home-list-chats]
- :<- [:view-id]
  :<- [:home-items-show-number]
- (fn [[chats view-id home-items-show-number]]
-   (if (or (empty? @memo-chats-stack-items) (= view-id :chats-stack))
-     (let [res (take home-items-show-number chats)]
-       (reset! memo-chats-stack-items res)
-       res)
-     ;;we want to keep data unchanged so react doesn't change component when we leave screen
```

Comment:
I'm not sure what this comment is referring to. React itself doesn't change the data, and re-frame just recompute subs so it'd reach the same value.

AFAIK, Re-reframe compares using Clojure's `=`, so we don't need a stable reference.

I don't understand the usage of the atom 🤔

Reply:
Hi @ulisesmac, when a user receives a message, the last-message key is updated, and the chat screen list item is refreshed.

If the user is not on the chat home screen (e.g., while chatting), this atom provides cached data to prevent unnecessary updates on the home screen.

I can see the usefulness of this, but it’s an older performance improvement, and @flexsurfer can probably provide more context.

Reply:
@Parveshdhull Thank you so much for the context!

I've always wondered what happens with the screens in a navigation stack that aren't focused. Are their subscriptions still mounted and running? by what you say, I infer that the answer is YES.

> when a user receives a message, the last-message key is updated, and the chat screen list item is refreshed.

I wonder what is the impact this may have, since it's just a rerender of a single chat item it might be very cheap, right? I mean, what is going on in the chat screen is a lot more complex/costly so the perf gain is probably negligible.

Thanks for the explanation @Parveshdhull!

Reply:
@ilmotta 

This fix made me think about the global atoms we use and if we are properly resetting all of them :thinking:

Reply:
Most importantly, don’t forget: everything might look perfect to you simply because no one is actively using the app and reporting how it lags. Just take an old device and start actively using the app — with hundreds of contacts and thousands of chats — and you’ll be unpleasantly surprised at how poorly and slowly the app performs.

Reply:
Fair points @flexsurfer 💯 Thanks for the context, this will be useful in my attempts to make some optimizations to the chat list rendering.